web/docs/internal/gestures/gesture-input-serialization.md

```diff
@@ -0,0 +1,25 @@
+# Gesture Input Serialization
+
+The GestureRecognizer engine is designed to handle and process asynchronous, continuous input of one or more touchpoints at the same time.  The processing of events takes time, though, and on older devices it is possible for new input events to come in while older events are still being processed.  As it is possible for an earlier event to affect how a later event should be processed, it is necessary to ensure that all events are processed in a specific order. We wish to capture the user's original intent for each aspect of input and later match it against the system state that a user would reasonably anticipate for the input based on their previous interactions, even if such a state hasn't yet actually been reached or activated at the true time of input.
+
+For a keyboard, suppose a user taps the `Shift` and `A` keys with minimal time gap between the two events.  The `Shift` state may not be in place yet once the `A` is tapped - event processing does take _some_ time, after all.  The serialization scheme used by the engine will delay processing for the `A` until the `Shift` state _has_ taken effect, leading a capital `A` to be emitted instead of a lowercase `a`.
+
+This serialization scheme used by the gesture engine is further complicated by the feature allowing a gesture to be resolved when receiving a new input _without_ associating that input with the gesture.  For example, gestures can be specified such that one tap's gesture instantly finalizes whenver newly-incoming second tap is received.  Should finalization of the former trigger a change in engine state - say, to model a change in layer - it is possible that the newly-incoming input should not be processed further until the former input is _fully_ processed.  Furthermore, the engine leverages `Promise`s internally to help keep the different layers of [gesture processing](./gesture-processing.md) separate and isolated; these need a chance to resolve before continuing processing of the new event and any events that may follow it.
+
+## Gesture Processing serialization
+
+The first point that requires special handling is detection of newly-received inputs - new mouse or touch interaction points.  As some gesture models may specify autocompletion whenever a new touch comes in, those should resolve before further handling the new input.
+
+Toward this end, `MatcherSelector`'s `matchGesture` method maintains a `Promise` serving as a lock within the field `pendingMatchSetup`.  This is set whenever checking such cases - whether or existing gestures have special interactions triggered by new touchpoints - and is cleared once these checks have completed.  This lock is referenced early within the same method to prevent `matchGesture` from handling any subsequent inputs while waiting on this process to complete.
```

Comment:
Is there a "not" missing?
```suggestion
Toward this end, `MatcherSelector`'s `matchGesture` method maintains a `Promise` serving as a lock within the field `pendingMatchSetup`.  This is set whenever checking such cases - whether or not existing gestures have special interactions triggered by new touchpoints - and is cleared once these checks have completed.  This lock is referenced early within the same method to prevent `matchGesture` from handling any subsequent inputs while waiting on this process to complete.
```